components/Modal.tsx

```diff
@@ -0,0 +1,51 @@
+import { FC, useState, useEffect } from 'react';
+import Sidebar from './SideBar';
+import NavBar from './NavBar';
+
+type ModalProps = {
+  show: boolean;
+  setShow: SetShowFunction;
+};
+
+type SetShowFunction = (show: boolean) => void;
+
+const Modal: FC<ModalProps> = ({ show, setShow }) => {
+  if (!show) {
+    return null;
+  }
+
+  return (
+    <div
+      className="flex items-center justify-center fixed inset-0 h-full w-full bg-ocf-black bg-opacity-50"
+      onClick={() => setShow(false)}
+    >
+      <div
+        className="w-80 h-auto bg-ocf-gray-1000 text-white opacity-100 px-8 py-6 rounded-lg"
+        onClick={(e: React.MouseEvent<HTMLInputElement>) => e.stopPropagation()}
+      >
```

Comment:
could we change the width of the modal to line up with the widths of the form input fields behind it?